packages/core/test/scheduler.test.ts

```diff
@@ -0,0 +1,31 @@
+import { createActor, createMachine } from '../src';
+
+describe('scheduler', () => {
+  it('Should be able to use a custom scheduler', (done) => {
+    const machine = createMachine({
+      id: 't',
+      after: {
+        1000: {}
+      }
+    });
+
+    const actor = createActor(machine, {
+      scheduler: {
+        setTimeout: (_system, scheduledEvent) => {
```

Comment:
if we are gonna keep those arguments i think we should rename `setTimeout`/`cancelTimeout` options on this. They are too close to the JS timers API and it's confusing that they accept different arguments here.

Any particular reason why you have restructured it this way? could we add a test case that would show why this is useful?